File: exp/models/task.js
const mongoose = require("mongoose");

const taskSchema = mongoose.Schema({
  content: String,
  priority: { type: Number, default: 0 },
  timestamp: { type: Date, default: Date.now },
  completed: { type: Boolean, default: false },
});

module.exports = mongoose.model("Task", taskSchema);

File: fpi/models/task.py
from __future__ import annotations

from typing import Optional
from datetime import datetime

from bson import ObjectId
from pydantic import BaseModel, Field


class PyObjectId(ObjectId):
    @classmethod
    def __get_validators__(cls):
        yield cls.validate

    @classmethod
    def validate(cls, v):
        if not ObjectId.is_valid(v):
            raise ValueError('Invalid objectid')
        return ObjectId(v)

    @classmethod
    def __modify_schema__(cls, field_schema):
        field_schema.update(type='string')


class Task(BaseModel):
    content: str
    priority: int = 1
    completed: bool = False
    id: Optional[PyObjectId] = Field(alias='_id')
    timestamp: Optional[datetime] = Field(default_factory=datetime.now)

    class Config:
        arbitrary_types_allowed = True
        json_encoders = {ObjectId: str}
